File: python/영역 구하기(2583) copy.py
# 첫 줄 입력 : M, N, K
# 둘째 줄 부터 K개의 줄
# 한 줄에 하나씩 왼쪽 아래(x,y), 오른쪽 위(x,y) 좌표
from collections import deque

m, n, k = tuple(map(int, input().split()))
graph = [[0] * n for _ in range(m)]
visited = [[False] * n for _ in range(m)]

dx = [-1, 1, 0, 0]
dy = [0, 0, -1, 1]
cnt_list = []

for _ in range(k):
    bottom_x, bottom_y, top_x, top_y = tuple(map(int, input().split()))

    for i in range(bottom_y, top_y):
        for j in range(bottom_x, top_x):
            # 직사각형 부분 1로 처리
            graph[i][j] = 1
    
def bfs(x,y):
    queue = deque()
    queue.append((x,y))
    visited[x][y] = True

    cnt = 1
    while queue:
        x, y = queue.popleft()

        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]

            if nx >=0 and nx < m and ny >=0 and ny < n and visited[nx][ny] == False and graph[nx][ny] == 0:
                visited[nx][ny] = True
                queue.append((nx,ny))
                cnt += 1
        
    cnt_list.append(cnt)
    
for i in range(m):
    for j in range(n):
        if graph[i][j] == 0 and visited[i][j] == False:
            bfs(i,j)

cnt_list.sort()
print(len(cnt_list))
for elem in cnt_list:
    print(elem, end=' ')

File: js/영역 구하기(2583) copy.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [m, n, k] = input[0].split(' ').map(Number);
const graph = Array.from({length: m}, () => Array(n).fill(0));
const visited = Array.from({length: m}, () => Array(n).fill(false));

const dx = [-1, 1, 0, 0];
const dy = [0, 0, -1, 1];

for (let i=1; i<k+1; i++) {
    const [bottom_x, bottom_y, top_x, top_y] = input[i].split(' ').map(Number);
    for (let x=bottom_y; x<top_y; x++) {
        for (let y=bottom_x; y<top_x; y++) {
            graph[x][y] = 1;
        }
    }
}

let cnt_list = [];

function bfs(x,y) {
    let queue = [];
    queue.push([x,y]);
    visited[x][y] = true;
    let cnt = 1;

    while (queue.length) {
        const [x,y] = queue.shift();

        for (let i=0; i<4; i++) {
            let nx = x + dx[i];
            let ny = y + dy[i];

            if (nx >=0 && nx < m && ny >=0 && ny < n && !visited[nx][ny] && graph[nx][ny] === 0) {
                visited[nx][ny] = true;
                queue.push([nx,ny]);
                cnt++;
            }
        }
    }

    cnt_list.push(cnt);
}

for (let i=0; i<m; i++) {
    for (let j=0; j<n; j++) {
        if (!visited[i][j] && graph[i][j] === 0) {
            bfs(i,j);
        }
    }
}

cnt_list.sort((a,b) => a-b);
console.log(cnt_list.length);
console.log(cnt_list.join(' '));